### ISSUE TITLE
[FEATURE] Weapon carousel

### ISSUE THREAD

**Issue by electricbrass** [odamex/odamex#1101]
**Is your feature request related to a problem? Please describe.**
Recent official ports as well as some others such as Woof have added a weapon carousel when playing on controller. It makes cycling through weapons a lot nicer.

![392685093-76ad9c26-a36b-4c9a-aa11-250c8501796c](https://github.com/user-attachments/assets/aebff443-fbd3-4d75-abfb-1a40643b2ad8)

**Describe the solution you'd like to see**
Carousel similar to those mentioned above, with graphics in odamex.wad.

**Describe alternatives you've considered**
Loading graphics from extras.wad, but this contains a lot of other extra lumps and may not always be present.

**Comment by rakohus** [comment]
I kind of feel like there's some use-case and behavioral overlap between this and #1074. 1074 was never finally settled but I think we could probably work toward a consensus with some more discussion.

This is a preference thing, but _personally_ I think using the numbers is better because it allows for it to be universal _across all wads_ that don't use the typical doom weapons. You want to load REKKR or Chex? You get the numbers instead of pictures of Doom weapons that don't exist in those wads. It's a utility vs aesthetic argument.

**Comment by electricbrass** [comment]
I agree that it's better to have something that fits all wads. I think the primary thing then is, what about when using the status bar and not the fullscreen hud? I personally would not want the numbers always displayed, so they should probably pop up either at the top of the screen or just above the status bar just when switching. Functionality for scrolling through the numbers to switch instead of trying to switch as soon as the button is pressed would also still be needed.

**Comment by electricbrass** [comment]
There is one reason I think to prefer the version with the graphics: I suspect we'll be seeing more wads in the future that include their own graphics for the weapon carousel. It remains to be seen how much adoption that part of ID24 will see, but it will probably see at least some use. And I think being able to display as many of the custom graphics from a wad as possible is something desirable. If someone puts in the work to make new icons to match their custom weapons, I want to see them.

**Comment by elf-alchemist** [comment]
>I suspect we'll be seeing more wads in the future that include their own graphics for the weapon carousel.

For reference, a newer release such as [Chex Quest 3: Vanilla Edition](https://melodic-spaceship.neocities.org/chex3v/downloads), has already shipped it's version 1.0 with carousel icons. Other than this one, many older classic wads have already received patches to include custom icons by Scuba Steve in the Community Curated Uploads thread on Doomworld. There's definitely an interest in this, already.

Screenshot from Woof.
![woof0039](https://github.com/user-attachments/assets/dae1ec3d-1726-45ff-9fae-94dfeec818e1)
